apps/agreement/contracts/Agreement.sol

```diff
@@ -761,21 +775,25 @@ contract Agreement is IAgreement, AragonApp {
             challengerFeeAmount
         );
 
-        // Pull arbitration fees from submitter
+        // Pull arbitration fees from submitter, note that if missing fees is zero this doesn't revert
         address submitter = _action.submitter;
-        _transferFrom(feeToken, submitter, missingFees);
+        _depositFrom(feeToken, submitter, missingFees);
 
         // Create dispute. The arbitrator should pull any arbitration fees from this Agreement here.
         // To be safe, We first set the allowance to zero in case there is a remaining approval for the arbitrator.
         // This is not strictly necessary for ERC20s, but some tokens, e.g. MiniMe (ANT and ANJ),
         // revert on an approval if an outstanding allowance exists
-        _approveArbitratorFeeTokens(feeToken, disputeFeeRecipient, 0);
-        _approveArbitratorFeeTokens(feeToken, disputeFeeRecipient, totalFees);
+        _approveFor(feeToken, disputeFeeRecipient, 0);
+        _approveFor(feeToken, disputeFeeRecipient, totalFees);
         uint256 disputeId = _arbitrator.createDispute(DISPUTES_POSSIBLE_OUTCOMES, _metadata);
 
-        // Return any remaining portion of the pre-paid arbitrator fees to the challenger, if necessary
         if (challengerFeeToken != feeToken) {
-            _transfer(challengerFeeToken, _challenge.challenger, challengerFeeAmount);
+            // Return any remaining portion of the pre-paid arbitrator fees to the challenger, if necessary
+            _transferTo(challengerFeeToken, _challenge.challenger, challengerFeeAmount);
+        } else if (missingFees == 0) {
```

Comment:
I wonder if we shouldn't use `challengerFeeAmount < totalFees` as the check instead, since that's what we're subtracting.

But another thought I had was to directly return a `challengerRefund` value from `_getMissingArbitratorFees()` instead.